packages/@aws-cdk/aws-codepipeline/test/pipeline.test.ts

```diff
@@ -485,6 +486,298 @@ describe('', () => {
+        context: app2,
+        suffix: '2',
+        stackId: 'STACK-ID',
+        nestedStackId: 'Nested',
+        pipelineId: 'PIPELINE-ID',
+      });
+
+      expect(Template.fromStack(stack1.nestedStack!).findResources(kmsAliasResource))
+        .toEqual(Template.fromStack(stack2.nestedStack!).findResources(kmsAliasResource));
+
+      Template.fromStack(stack1.nestedStack!).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-stackidnestedpipelineid3e91360a',
+      });
+
+      Template.fromStack(stack2.nestedStack!).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-stackidnestedpipelineid3e91360a',
+      });
+    });
+
+    testFutureBehavior('cross account key alias names do not conflict for nested stacks when in the same stack but nested stacks have different IDs when feature flag is enabled', { [cxapi.CODEPIPELINE_CROSS_ACCOUNT_KEY_ALIAS_STACK_SAFE_UNIQUE_ID]: true }, cdk.App, (app) => {
+      const stack = createPipelineStack({
+        context: app,
+        suffix: 'Name-1',
+        stackId: 'STACK-ID',
+        nestedStackId: 'First',
+        pipelineId: 'PIPELINE-ID',
+      });
+      const nestedStack2 = new cdk.NestedStack(stack, 'Second');
+      createPipeline(nestedStack2, 'Actual-Pipeline-Name-2', 'PIPELINE-ID');
+
+      expect(Template.fromStack(stack.nestedStack!).findResources(kmsAliasResource))
+        .not.toEqual(Template.fromStack(nestedStack2).findResources(kmsAliasResource));
+
+      Template.fromStack(stack.nestedStack!).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-actualstackname1firstpipelineid3c59cb88',
+      });
+
+      Template.fromStack(nestedStack2).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-actualstackname1secondpipelineid16143d12',
+      });
+    });
+
+    testLegacyBehavior('cross account key alias names do not conflict for nested stacks when in the same stack but nested stacks have different IDs when feature flag is not enabled', cdk.App, (app) => {
+      const stack = createPipelineStack({
+        context: app,
+        suffix: 'Name-1',
+        stackId: 'STACK-ID',
+        nestedStackId: 'First',
+        pipelineId: 'PIPELINE-ID',
+      });
+      const nestedStack2 = new cdk.NestedStack(stack, 'Second');
+      createPipeline(nestedStack2, 'Actual-Pipeline-Name-2', 'PIPELINE-ID');
+
+      expect(Template.fromStack(stack.nestedStack!).findResources(kmsAliasResource))
+        .not.toEqual(Template.fromStack(nestedStack2).findResources(kmsAliasResource));
+
+      Template.fromStack(stack.nestedStack!).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-stackidfirstpipelineid5abca693',
+      });
+
+      Template.fromStack(nestedStack2).hasResourceProperties(kmsAliasResource, {
+        AliasName: 'alias/codepipeline-stackidsecondpipelineid288ce778',
+      });
+    });
+  });
```

Comment:
Thoughts about putting this into a separate test file? Doesn't seem like it's much related to the other tests in this file.

Reply:
I'm torn about this because it is part of creating the pipeline, but also, this file is about a million lines long.